<!--/*****************************************************************
 * labust_navigation_template.launch
 *
 *  Created on: Apr 16, 2014
 *      Author: Filip Mandic
 *
 ********************************************************************/

/*********************************************************************
* Software License Agreement (BSD License)
*
*  Copyright (c) 2014, LABUST, UNIZG-FER
*  All rights reserved.
*
*  Redistribution and use in source and binary forms, with or without
*  modification, are permitted provided that the following conditions
*  are met:
*
*   * Redistributions of source code must retain the above copyright
*     notice, this list of conditions and the following disclaimer.
*   * Redistributions in binary form must reproduce the above
*     copyright notice, this list of conditions and the following
*     disclaimer in the documentation and/or other materials provided
*     with the distribution.
*   * Neither the name of the LABUST nor the names of its
*     contributors may be used to endorse or promote products derived
*     from this software without specific prior written permission.
*
*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
*  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
*  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
*  POSSIBILITY OF SUCH DAMAGE.
******************************************************************/-->

<launch>

    <!-- ========================================================= -->
    <!-- == Arguments ============================================ -->
    <!-- ========================================================= -->

    <rosparam command="load" file="$(find labust_mission)/launch/labust_mission/config/navparams_ex.yaml" />

    <arg name="ekfAbs" default="false" />

    <!-- ========================================================= -->
    <!-- == Navigation filter node =============================== -->
    <!-- ========================================================= -->

     <node 
        pkg="labust_navigation" 
        type="ekf3d_node" 
        name="ekf_nav">

	<param name="imu_with_yaw_rate" value="true" />
	<param name="dvl_model" value="1" />
	<param name="Ts" value="0.1" />
        <!--<param name="absoluteEKF" value="false" />-->

    </node> 

    <node 
        if="$(arg ekfAbs)"
        pkg="labust_navigation" 
        type="ekf3d_node" 
        name="ekf_nav_absolute">

	<param name="imu_with_yaw_rate" value="true" />
	<param name="dvl_model" value="1" />
	<param name="Ts" value="0.1" />
        <param name="absoluteEKF" value="true" />
	<!--<remap from="depth" to="no_depth" />-->
	 <remap from="stateHat" to="stateHatAbs" /> 

    </node>

    <!-- ========================================================= -->
    <!-- == LocalFixSim node ===================================== -->
    <!-- ========================================================= -->

    <node 
        pkg="labust_navigation" 
        type="llnode" 
        name="tf_frame_node">

        <param name="LocalFixSim" value="true" if="$(optenv USE_LOCAL_FIX 1)" />

    </node>

</launch>
